### ISSUE TITLE
algo add problem!

### ISSUE THREAD

**Issue by MTCHANNELL** [ROZ-MOFUMOFU-ME/zny-nomp#55]
Hello. I'm trying to add the "timetravel" algorithm to the pool. but I could not succeed. The error I get is as follows.
We thought a block was found but it was rejected by the daemon, share data: {"job":"1","ip":"::ffff:192.168.0.30","port":3032,"worker":"9LiGvrXXwUBh49xCfrV1GmBAFufXUrdvxs","height":1725010,"blockReward":281250000,"difficulty":0.5,"shareDiff":"359.74194656","blockDiff":287.355439872,"blockDiffActual":1.122482187,"blockHash":"00000000b62c189ba71940c292075d4e62da204699be19b9c2aa81141ddadd22","blockHashInvalid":"00000000b62c189ba71940c292075d4e62da204699be19b9c2aa81141ddadd22"}

I adjusted the difficulty settings many times, but I don't think that's the problem. I would be very pleased if you could help me. I will post links to the files I edited later. I would be very grateful if you could check it or guide me. I wish you good work.

**Comment by emerauda** [comment]
Thank you for submitting your issue. The error message you've shared is insufficient for a complete diagnosis. The concept of "timetravel" in this context is not explained, which makes it difficult to provide assistance. It's important to include more context and details in your report to help others understand your issue better.
Before seeking further help, I strongly recommend thoroughly reviewing the relevant documentation and ensuring that all configurations are set correctly. If specific details such as error logs, code snippets, and configurations are not provided soon, this issue may be closed without further investigation.

**Comment by MTCHANNELL** [comment]
Ok. I will explain to you in detail everything I did.
01-) redis.conf edit;
bind 192.168.0.14
port 6379

02-) machinecoin.conf edit;
rpcuser=mtchannelMAC
rpcpassword=mtchannelMAC
rpcbind=192.168.0.14
rpcport=9998
rpcallowip=192.168.0.0/24
server=1
listen=1

03-) config.json edit;
[config.json](https://github.com/user-attachments/files/17734432/config.json)

04-) pool_configs folder edit;
[timetravel.json](https://github.com/user-attachments/files/17734438/timetravel.json)

05-) paymentProcessor.js edit;
"ismine" change "isvalid"

06-) coins folder edit;
[timetravel.json](https://github.com/user-attachments/files/17734453/timetravel.json)

07-) folder edit;
multi-hashing edit;
binding gyp;
add;  "src/timetravel.c",
src folder add;
timetravel.c
timetravel.h
[multihashing.json](https://github.com/user-attachments/files/17734484/multihashing.json)

stratum-pool edit;
algoProperties.js edit;
    },
    timetravel: {
        multiplier: 256,
        hash: function (coinConfig) {
            var nTimestamp = coinConfig.chainStartTime || 1731465538;
            return function (data, nTime) {
				return multiHashing.timetravel(data, nTime, nTimestamp);
            }
        }
    },

So far I've only added and changed these. but when I start the pool I don't get any errors. I see the following error only when the block is found.
[Pool]	[machine] We thought a block was found but it was rejected by the daemon, share data: {"job":"a","ip":"::ffff:192.168.0.30","port":3032,"worker":"9LiGvrXXwUBh49xCfrV1GmBAFufXUrdvxs","height":1725805,"blockReward":281250000,"difficulty":9.36329588,"shareDiff":"3201.17586565","blockDiff":1239.605764608,"blockDiffActual":4.842210018,"blockHash":"000000001478dfcb7d9818c4ee962c37de96cb3dfbec8493b8eda0038b0ef2c8","blockHashInvalid":"000000001478dfcb7d9818c4ee962c37de96cb3dfbec8493b8eda0038b0ef2c8"}

./machinecoin-cli -rpcconnect=192.168.0.14 -rpcpassword=mtchannelMAC -rpcport=9998 -rpcuser=mtchannelMAC getwalletinfo
{
  "walletname": "",
  "walletversion": 169900,
  "balance": 0.00000000,
  "unconfirmed_balance": 0.00000000,
  "immature_balance": 14.06250000,

I tested it for 3-4 hours. "immature_balance" is accumulating but I cannot receive payment. It does not transfer to "balance".

![image](https://github.com/user-attachments/assets/d3a4bd74-383d-42d0-b6a4-5dc20377e132)

![image](https://github.com/user-attachments/assets/b9383d93-ee44-4218-8820-bf165d2a03e9)

If there is any information you need that I don't remember, I would appreciate it if you let me know. But for now I only remember that I did these.

**Comment by MTCHANNELL** [comment]
I use the wallet as "legacy" for the pool. I also tried it on other wallet types to test it out. But I got the same result in all of them.
./machinecoind -datadir=/root/Masaüstü/MACHiNE/wallet -addresstype="legacy"
./machinecoin-cli -rpcconnect=192.168.0.14 -rpcpassword=2502MtChannel2502 -rpcport=9998 -rpcuser=mtchannel getnewaddress

miner command line;
wildrig-v0.39.0.exe -a timetravel -o stratum+tcp://192.168.0.14:3032 -u 9LiGvrXXwUBh49xCfrV1GmBAFufXUrdvxs --print-time 3

**Comment by MTCHANNELL** [comment]
before;
./machinecoin-cli -rpcconnect=192.168.0.14 -rpcpassword=mtchannelMAC -rpcport=9998 -rpcuser=mtchannelMAC getwalletinfo
{
"walletname": "",
"walletversion": 169900,
"balance": 0.00000000,
"unconfirmed_balance": 0.00000000,
"immature_balance": 14.06250000,

after;
./machinecoin-cli -rpcconnect=192.168.0.14 -rpcpassword=mtchannelMAC -rpcport=9998 -rpcuser=mtchannelMAC getwalletinfo
{
"walletname": "",
  "walletversion": 169900,
  "balance": 33.75000000,
  "unconfirmed_balance": 0.00000000,
  "immature_balance": 0.00000000,
  
  I would like to inform you about the latest situation. When I check the wallet hours later, I can see that the "immature_balance" balances have been transferred to the "balance" section.

01-) The pool does not automatically send money to the wallet from which I mined. The balance remains in the pool wallet. If I send it via manual command line with machinecoin-cli, I can send it to my miner wallet. But I don't know why it doesn't send automatically?
02-) When it finds a block, it continues to give the same error. I haven't found a solution yet myself. However, an error does not prevent payment to the pool wallet. I understand this.
03-) The website is not very important for me, but I would like to point out this. When I set up the pool and start mining. I can see miner on the website. However, sections such as "immature / balance / paid" always show zero. As I said, I don't usually use websites. API is sufficient for me. Just thought I'd let you know.
